inc/template.php

```diff
@@ -53,35 +92,32 @@ function get_template_name() : string {
 	return $template_name;
 }
 
-function load_template_or_404( string $template_name ) : void {
-	// Allow adding to the template paths, with the included demo ones as a fallback.
+/**
+ * Load a template or return 404 if it's not found. This is a copy of the WordPress `load_template` function, but with
+ * 404 handling.
+ *
+ * @param string $template_name Name of the template to load.
+ *
+ * @return bool Whether a template was loaded
+ */
+function load_htmx_template( string $template_name ) : bool {
+	// Allow adding to the template paths.
+	// phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores -- Established filter.
 	$paths = apply_filters( 'htmx.template_paths', [] );
-	if (empty($paths)) {
-		$paths[] = dirname( __FILE__, 2 ) . "/templates/";
-	}
 	array_walk( $paths, static function ( &$path ) {
 		$path = trailingslashit( $path );
 	} );
 
-	$match = false;
-	// match one path
+	// walk through the paths to find the first matching template.
+	$template_loaded = false;
 	foreach ( $paths as $path ) {
 		$path = "$path${template_name}.php";
 		if ( file_exists( $path ) ) {
-			$match = true;
-			$is_partial = str_starts_with( $template_name, 'partial-' );
-			if ( $is_partial ) {
-				include $path;
-				break;
-			}
+			$template_loaded = true;
 			load_template( $path );
 			break;
 		}
 	}
 
-	global $wp_query;
-	if ( ! $match ) {
-		$wp_query->set_404();
-		status_header( 404 );
-	}
+	return $template_loaded;
```

Comment:
I'm not testing for 404 here but why are there two places where templates are loaded, can this be combined?